Library/Homebrew/diagnostic.rb

```diff
@@ -566,20 +566,30 @@ def check_casktap_git_origin
         examine_git_origin(cask_tap.path, cask_tap.remote)
       end
 
-      def check_coretap_git_branch
+      sig { returns(T.nilable(String)) }
+      def check_tap_git_branch
         return if ENV["CI"]
+        return unless Utils::Git.available?
 
-        coretap_path = CoreTap.instance.path
-        return if !Utils::Git.available? || !(coretap_path/".git").exist?
+        commands = Tap.map do |tap|
+          next unless tap.path.git?
+          next if tap.path.git_origin.nil?
 
-        branch = coretap_path.git_branch
-        return if branch.blank? || branch.include?("master")
+          branch = tap.path.git_branch
+          next if branch.blank?
 
-        <<~EOS
-          #{CoreTap.instance.full_name} is not on the master branch.
+          origin_branch = Utils::Git.origin_branch(tap.path)&.split("/")&.last
```

Comment:
This is a bit hard to follow and feels a bit brittle. Might be worth being a dedicated (tested) function on `Utils::Git` or `Tap`

Reply:
I have a refactor for this written already but didn't open it yet because it would have caused trouble for this pull request.

Reply:
Fine with me if there's a refactor incoming 👍🏻